app/controllers/application_controller.rb

```diff
@@ -1,11 +1,17 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  class EmailNotFoundError < StandardError; end
+
   default_form_builder GOVUKDesignSystemFormBuilder::FormBuilder
 
   before_action :configure_permitted_parameters, if: :devise_controller?
   before_action :set_current_user
 
+  rescue_from EmailNotFoundError do
```

Comment:
I've had to add this in here as the before_action :set_current_user above is triggering the authentication workflow rather than on th intended request to /users/sign_in

Reply:
I don't quite understand. I thought `current_user` from devise would give me either the currently logged in user, or nil if there isn't any. Does it have a side effect of triggering the authentication flow?

Is so, maybe we want to wrap it in a `user_signed_in?` check. 

I really don't like this rescue here, since really, we only want to show it in a very specific case of user trying to sign in and the email not existing.

Reply:
@ethanmills @kicferk1 no @current_user = current_user if user_signed_in? throws the same error. Anyway, why do we need to set_current_user in application_controller?

Reply:
@ethanmills @kicferk1 I found something that works can you have another look please.

Reply:
Looks like the `raise` is still there? Now that we have the `unless :devise_controller?` can we not just render the  `email_not_found` directly?

Reply:
So I want to set current user as high up as possible, because I think vast majority of our app will behave differently for different users. And I think it is really convenient to be able to access current user from any view, without worrying that it will inadvertently throw an exception :/

Reply:
> Looks like the `raise` is still there? Now that we have the `unless :devise_controller?` can we not just render the `email_not_found` directly

@ethanmills I think you are looking at stale code

Reply:
Although, I am perfectly happy to not do it for devise controllers :)

Reply:
I'm looking at 271e78ce which has `raise Users::SessionsController::EmailNotFoundError`

Reply:
When the user email exists we render the flash message that the user should check their email. We are rendering email_not_found directly but only when the exception is raised. Not sure, if I have completely understood your point. @ethanmills

Reply:
I think this is fine now, sorry